shop4All-backend/src/main/java/com/example/shop4All_backend/Service/UserService.java

```diff
@@ -19,10 +23,28 @@ public class UserService {
     @Autowired
     private RoleRepo roleRepo;
 
-    public User registerNewUser(User user) {
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    //user registration(seller in this case)
+    public User registernewSeller(User user) throws Exception{
+        Optional<User> existingUser = userRepo.findById(user.getUserEmail());
+        if (existingUser.isPresent()) {
+            System.out.println("DSADASDa");
+            // Handle the case where the user already exists
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "User with this email already exists");
+        }
+
+        Role role = roleRepo.findById("Seller").orElseThrow(() -> new RuntimeException("Role not found"));
+        Set<Role> roles = new HashSet<>();
+        roles.add(role);
+        user.setRole(roles);
+
+        user.setUserPassword(getEncodedPassword(user.getUserPassword()));
         return userRepo.save(user);
     }
```

Comment:
Naming convention for this method should be "registerNewSeller".
Be careful at the indentation: "throws Exception {"

If you want to use logs, use logger instead of prints.